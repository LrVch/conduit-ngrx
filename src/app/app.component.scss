// @import '../styles-variables';

.theme-wrapper {
  width: 100%;
  height: 100%;
}

mat-sidenav-container {
  height: 100%;
  width: 100%;
}

mat-sidenav {
  width: 250px;

  // .mat-nav-list {
  //   padding-top: 0;
  // }
}