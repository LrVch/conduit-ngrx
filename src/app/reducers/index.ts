import {
  ActionReducerMap,
  MetaReducer
} from '@ngrx/store';
import { environment } from '@env/environment';
import { AuthState, authReducer } from '@app/auth/auth.reducer';
import { ArticlesState, articlesReducer } from '@app/articles/articles.reducer';
import { storeFreeze } from 'ngrx-store-freeze';
import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { ArticlesConfigState, articlesConfigReducer } from '@app/articles/articlesConfig.reducer';
import { ProfileState, profileReducer } from '@app/profile/profile.reducer';
import { layoutReducer, LayoutState } from '@app/layout/layout.reducer';
import { EditorState, editorReducer } from '@app/editor/editor.reducer';
import { ArticleState, articleReducer } from '@app/article/article.reducer';
import { AppSettingsState } from '@app/core/models/app-settings.model';
import { appSettingsReducer } from '@app/appSettings/app-settings.reducer';
import { debug } from './meta-reducers/debug.reducer';

import { initStateFromLocalStorage } from './meta-reducers/init-state-from-local-storage.reducer';


export interface AppState {
  auth: AuthState;
  articles: ArticlesState;
  router: RouterReducerState;
  articlesConfig: ArticlesConfigState;
  profile: ProfileState;
  layout: LayoutState;
  editor: EditorState;
  article: ArticleState;
  appSettings: AppSettingsState;
}

export const reducers: ActionReducerMap<AppState> = {
  auth: authReducer,
  articles: articlesReducer,
  router: routerReducer,
  articlesConfig: articlesConfigReducer,
  profile: profileReducer,
  layout: layoutReducer,
  editor: editorReducer,
  article: articleReducer,
  appSettings: appSettingsReducer
};


export const metaReducers: MetaReducer<AppState>[] = [initStateFromLocalStorage];

if (!environment.production) {
  metaReducers.unshift(storeFreeze);
}
