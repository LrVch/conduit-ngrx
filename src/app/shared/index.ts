export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './shared.module';
export * from './utils';
export * from './list-error/list-error.component';
export * from './show-authed.directive';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
