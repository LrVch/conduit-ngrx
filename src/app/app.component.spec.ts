import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from './lib/testing';
import { AppComponent } from './app.component';
import {
  DebugElement,
  Component,
  Input,
  Output,
  EventEmitter
} from '@angular/core';
import { MainLoaderService } from './core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { Store, StoreModule, combineReducers } from '@ngrx/store';
import * as fromRoot from './reducers';
import * as fromAuth from './auth/auth.reducer';
import { RouterTestingModule } from '@angular/router/testing';
import {
  ShowMainLoader,
  HideMainLoader,
  LayoutActionTypes,
  ToggleSideNav
} from './layout/layout.actions';
import { cold } from 'jasmine-marbles';
import { Subject } from 'rxjs';
import { MaterialModule } from './shared';
import {
  AppSettingsChangeLanguage,
  AppSettingsChangeStickyHeader,
  AppSettingsChangeTheme,
  AppSettingsChangeHour,
  AppSettingsChangeAutoNightMode,
  AppSettingsChangeAsideOpenMode,
  AppSettingsChangeNightModeFrom,
  AppSettingsChangeNightModeTo,
  AppSettingsChangeRouteAnimationType,
  AppSettingsChangeRouteAnimationEnabled
} from './appSettings/app-settings.actions';
import { By } from '@angular/platform-browser';
import {
  DEFAULT_THEME,
  BLACK_THEME,
  BLUE_THEME
} from './core/models/app-settings.model';

const themes = [DEFAULT_THEME, BLUE_THEME, BLACK_THEME];

const themesViewValueMap = {
  [DEFAULT_THEME]: { value: 'purple', color: '#673ab7' },
  [BLACK_THEME]: { value: 'black', color: '#616161' },
  [BLUE_THEME]: { value: 'blue', color: '#1976d2' }
};

@Component({
  selector: 'app-header',
  template: `
    <div id="lang" (click)="onchangeLanguage()"></div>
    <div id="toggleNav" (click)="ontoggleSideNave()"></div>
  `
})
class TestHostHeaderComponent {
  @Input('user') user: any;
  @Input('loggedIn') loggedIn: any;
  @Input('showLoader') showLoader: any;
  @Input('languages') languages: any;
  @Input('language') language: any;
  @Output() changeLanguage = new EventEmitter();
  @Output() toggleSideNave = new EventEmitter();

  onchangeLanguage() {
    this.changeLanguage.emit('ru');
  }

  ontoggleSideNave() {
    this.toggleSideNave.emit();
  }
}

@Component({
  selector: 'app-footer',
  template: ``
})
class TestHostFooterComponent {}

@Component({
  selector: 'app-aside',
  template: `
    <div id="changeAnimType" (click)="onchangeRouteAnimationsType()"></div>
    <div id="stikyHeader" (click)="onchangeStickyHeader()"></div>
    <div id="asideOpenMode" (click)="onchangeAsideOpenMode()"></div>
    <div id="autoNightMode" (click)="onchangeAutoNightMode()"></div>
    <div id="from" (click)="onchangeAutoNightModeFrom()"></div>
    <div id="to" (click)="onchangeAutoNightModeTo()"></div>
    <div id="animationEnabled" (click)="onchangeRouteAnimationsEnable()"></div>
    <div id="changeTheme" (click)="onchangeTheme()"></div>
  `
})
class TestHostAisdeComponent {
  @Input('theme') theme: string;
  @Input('themes') themes: any;
  @Input('stickyHeader') stickyHeader: boolean;
  @Input('routeAnimationsEnabled') routeAnimationsEnabled: boolean;
  @Input('asideOpenMode') asideOpenMode: any;
  @Input('asideOpenModes') asideOpenModes: any;
  @Input('autoNightMode') autoNightMode: string;
  @Input('autoNightModeFrom') autoNightModeFrom: any;
  @Input('autoNightModeTo') autoNightModeTo: any;
  @Input('routeAnimateonsChangeType') routeAnimateonsChangeType: any;
  @Input('routeAnimateonsChangeTypes') routeAnimateonsChangeTypes: any;

  @Output() changeRouteAnimationsType = new EventEmitter();
  @Output() changeStickyHeader = new EventEmitter();
  @Output() changeAsideOpenMode = new EventEmitter();
  @Output() changeAutoNightMode = new EventEmitter();
  @Output() changeAutoNightModeFrom = new EventEmitter();
  @Output() changeAutoNightModeTo = new EventEmitter();
  @Output() changeRouteAnimationsEnable = new EventEmitter();
  @Output() changeTheme = new EventEmitter();

  onchangeRouteAnimationsType() {
    this.changeRouteAnimationsType.emit('SLIDE');
  }

  onchangeStickyHeader() {
    this.changeStickyHeader.emit(false);
  }

  onchangeAsideOpenMode() {
    this.changeAsideOpenMode.emit('push');
  }

  onchangeAutoNightMode() {
    this.changeAutoNightMode.emit(true);
  }

  onchangeAutoNightModeFrom() {
    this.changeAutoNightModeFrom.emit(14);
  }

  onchangeAutoNightModeTo() {
    this.changeAutoNightModeTo.emit(7);
  }

  onchangeRouteAnimationsEnable() {
    this.changeRouteAnimationsEnable.emit(false);
  }

  onchangeTheme() {
    this.changeTheme.emit('BLACK-THEME');
  }
}

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let de: DebugElement;
  let store: Store<fromAuth.AuthState>;
  let mainLoaderService: MainLoaderService;
  let resutlThemes: any[];

  class MockMainLoaderService {
    showLoader = new Subject<boolean>();
    showLoader$ = this.showLoader.asObservable();
  }

  beforeEach(async(() => {
    resutlThemes = themes.map(theme => {
      return {
        value: theme.toLowerCase(),
        viewValue: `conduit.menu.theme.${themesViewValueMap[theme].value}`,
        color: themesViewValueMap[theme].color
      };
    });
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [
          AppComponent,
          TestHostHeaderComponent,
          TestHostFooterComponent,
          TestHostAisdeComponent
        ],
        imports: [
          NoopAnimationsModule,
          MaterialModule,
          StoreModule.forRoot({
            ...fromRoot.reducers,
            feature: combineReducers(fromAuth.authReducer)
          }),
          RouterTestingModule
        ],
        providers: [
          { provide: MainLoaderService, useClass: MockMainLoaderService }
        ]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(AppComponent);
      component = fixture.componentInstance;
      de = fixture.debugElement;
      store = testBed.get(Store);
      mainLoaderService = testBed.get(MainLoaderService);
      fixture.detectChanges();

      spyOn(store, 'dispatch').and.callThrough();
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile', () => {
    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should dispatch "ShowMainLoader"', () => {
    mainLoaderService.showLoader.next(true);
    expect(store.dispatch).toHaveBeenCalledWith(new ShowMainLoader());
  });

  it('should dispatch "HideMainLoader"', () => {
    mainLoaderService.showLoader.next(false);
    expect(store.dispatch).toHaveBeenCalledWith(new HideMainLoader());
  });

  it('should get showMailLoader', () => {
    component.ngOnInit();

    const expected = cold('a', { a: false });

    expect(component.showMailLoader$).toBeObservable(expected);
  });

  it('should show aside', () => {
    store.dispatch(new ToggleSideNav());
    component.ngOnInit();

    fixture.detectChanges();

    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should sticky header', () => {
    component.ngOnInit();
    store.dispatch(new AppSettingsChangeStickyHeader({ stickyHeader: true }));

    fixture.detectChanges();

    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should change theme to black', () => {
    component.ngOnInit();
    store.dispatch(new AppSettingsChangeTheme({ theme: 'BLACK-THEME' }));

    fixture.detectChanges();

    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should get loggedIn', () => {
    component.ngOnInit();

    const expected = cold('a', { a: false });

    expect(component.loggedIn$).toBeObservable(expected);
  });

  it('should get user', () => {
    component.ngOnInit();

    const expected = cold('a', { a: null });

    expect(component.user$).toBeObservable(expected);
  });

  it('should get languages', () => {
    component.ngOnInit();

    const expected = cold('a', {
      a: [{ value: 'en', viewValue: 'en' }, { value: 'ru', viewValue: 'ru' }]
    });

    expect(component.languages$).toBeObservable(expected);
  });

  it('should get language', () => {
    component.ngOnInit();

    const expected = cold('a', { a: '' });

    expect(component.language$).toBeObservable(expected);

    store.dispatch(new AppSettingsChangeLanguage({ language: 'ru' }));

    const expected2 = cold('a', { a: 'ru' });

    expect(component.language$).toBeObservable(expected2);
  });

  it('should get sideNavOpen', () => {
    component.ngOnInit();

    const expected = cold('a', { a: false });

    expect(component.sideNavOpen$).toBeObservable(expected);
  });

  it('should get effectiveTheme', () => {
    component.ngOnInit();
    store.dispatch(new AppSettingsChangeAutoNightMode({ autoNightMode: true }));
    store.dispatch(new AppSettingsChangeHour({ hour: 22 }));

    const expected = cold('a', { a: 'black-theme' });

    expect(component.effectiveTheme$).toBeObservable(expected);
  });

  it('should get theme', () => {
    component.ngOnInit();

    const expected = cold('a', { a: 'default-theme' });

    expect(component.theme$).toBeObservable(expected);
  });

  it('should get themes', () => {
    component.ngOnInit();
    const expected = cold('a', {
      a: resutlThemes
    });

    expect(component.themes$).toBeObservable(expected);
  });

  it('should get stickyHeader', () => {
    component.ngOnInit();

    const expected = cold('a', { a: true });

    expect(component.stickyHeader$).toBeObservable(expected);
  });

  it('should get asideOpenMode', () => {
    component.ngOnInit();

    const expected = cold('a', { a: 'push' });

    expect(component.asideOpenMode$).toBeObservable(expected);
  });

  it('should get asideOpenModes', () => {
    component.ngOnInit();

    const expected = cold('a', {
      a: [
        {
          value: 'push',
          viewValue: `conduit.menu.general.asideOpen.push`
        },
        {
          value: 'over',
          viewValue: `conduit.menu.general.asideOpen.over`
        }
      ]
    });

    expect(component.asideOpenModes$).toBeObservable(expected);
  });

  it('should get autoNightMode', () => {
    component.ngOnInit();

    const expected = cold('a', { a: false });

    expect(component.autoNightMode$).toBeObservable(expected);
  });

  it('should get autoNightModeFrom', () => {
    component.ngOnInit();

    const expected = cold('a', { a: 21 });

    expect(component.autoNightModeFrom$).toBeObservable(expected);
  });

  it('should get autoNightModeTo', () => {
    component.ngOnInit();

    const expected = cold('a', { a: 7 });

    expect(component.autoNightModeTo$).toBeObservable(expected);
  });

  it('should get routeAnimateonsChangeType', () => {
    component.ngOnInit();

    const expected = cold('a', { a: 'SLIDE' });

    expect(component.routeAnimateonsChangeType$).toBeObservable(expected);
  });

  it('should get routeAnimateonsChangeTypes', () => {
    component.ngOnInit();

    const expected = cold('a', {
      a: [
        {
          value: 'SLIDE',
          viewValue: `conduit.menu.animations.route.type.${'SLIDE'.toLocaleLowerCase()}`
        },
        {
          value: 'FADE',
          viewValue: `conduit.menu.animations.route.type.${'FADE'.toLocaleLowerCase()}`
        }
      ]
    });

    expect(component.routeAnimateonsChangeTypes$).toBeObservable(expected);
  });

  it('should dispathch "AppSettingsChangeLanguage"', () => {
    const lang = de.query(By.css('#lang'));
    lang.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeLanguage({ language: 'ru' })
    );
  });

  it('should dispathch "ToggleSideNav"', () => {
    const nav = de.query(By.css('#toggleNav'));
    nav.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(new ToggleSideNav());
  });

  it('should dispathch "ToggleSideNav"', () => {
    const nav = de.query(By.css('#toggleNav'));
    fixture.detectChanges();
    nav.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(new ToggleSideNav());
    fixture.detectChanges();
    const drop = de.query(By.css('.mat-drawer-backdrop'));
    drop.triggerEventHandler('click', null);

    expect(store.dispatch).toHaveBeenCalledWith(new ToggleSideNav());
  });

  it('should dispathch "AppSettingsChangeTheme"', () => {
    const button = de.query(By.css('#changeTheme'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeTheme({ theme: 'BLACK-THEME' })
    );
  });

  it('should dispathch "AppSettingsChangeStickyHeader"', () => {
    const button = de.query(By.css('#stikyHeader'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeStickyHeader({ stickyHeader: false })
    );
  });

  it('should dispathch "AppSettingsChangeAsideOpenMode"', () => {
    const button = de.query(By.css('#asideOpenMode'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeAsideOpenMode({ asideOpenMode: 'push' })
    );
  });

  it('should dispathch "AppSettingsChangeAutoNightMode"', () => {
    const button = de.query(By.css('#autoNightMode'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeAutoNightMode({ autoNightMode: true })
    );
  });

  it('should dispathch "AppSettingsChangeNightModeFrom"', () => {
    const button = de.query(By.css('#from'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeNightModeFrom({ nightModefrom: 14 })
    );
  });

  it('should dispathch "AppSettingsChangeNightModeTo"', () => {
    const button = de.query(By.css('#to'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeNightModeTo({ nightModeto: 7 })
    );
  });

  it('should dispathch "AppSettingsChangeRouteAnimationType"', () => {
    const button = de.query(By.css('#changeAnimType'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeRouteAnimationType({
        routeAnimationsChangeType: 'SLIDE'
      })
    );
  });

  it('should dispathch "AppSettingsChangeRouteAnimationEnabled"', () => {
    const button = de.query(By.css('#animationEnabled'));
    fixture.detectChanges();
    button.triggerEventHandler('click', null);
    expect(store.dispatch).toHaveBeenCalledWith(
      new AppSettingsChangeRouteAnimationEnabled({
        routeAnimationEnabled: false
      })
    );
  });
});
