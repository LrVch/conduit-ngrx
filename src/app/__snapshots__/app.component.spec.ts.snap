// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should change theme to black 1`] = `
<app-root
  asideOpenMode$={[Function Store]}
  asideOpenModes$={[Function Store]}
  autoNightMode$={[Function Store]}
  autoNightModeFrom$={[Function Store]}
  autoNightModeTo$={[Function Store]}
  effectiveTheme$={[Function Store]}
  language$={[Function Store]}
  languages$={[Function Store]}
  loggedIn$={[Function Store]}
  mainLoaderService={[Function MockMainLoaderService]}
  ngZone={[Function NgZone]}
  routeAnimateonsChangeType$={[Function Store]}
  routeAnimateonsChangeTypes$={[Function Store]}
  routeAnimationsEnabled$={[Function Store]}
  showMailLoader$={[Function Store]}
  sideNavOpen$={[Function Store]}
  stickyHeader$={[Function Store]}
  store={[Function Store]}
  theme$={[Function Store]}
  themes$={[Function Store]}
  user$={[Function Store]}
>
  <div
    class="ng-tns-c21-0 sticky-header theme-wrapper black-theme"
    ng-reflect-klass="theme-wrapper black-theme"
    ng-reflect-ng-class="[object Object]"
  >
    <mat-sidenav-container
      class="mat-drawer-container mat-sidenav-container"
    >
      
      <div
        class="mat-drawer-backdrop ng-star-inserted"
      />
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav
        class="mat-drawer mat-sidenav ng-tns-c23-1 ng-trigger ng-trigger-transform ng-tns-c21-0 mat-drawer-push ng-star-inserted"
        ng-reflect-mode="push"
        ng-reflect-opened="false"
        style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <app-aside
            class="ng-tns-c21-0"
            ng-reflect-aside-open-mode="push"
            ng-reflect-aside-open-modes="[object Object],[object Object"
            ng-reflect-auto-night-mode="false"
            ng-reflect-auto-night-mode-from="21"
            ng-reflect-auto-night-mode-to="7"
            ng-reflect-route-animateons-change-type="SLIDE"
            ng-reflect-route-animateons-change-types="[object Object],[object Object"
            ng-reflect-route-animations-enabled="true"
            ng-reflect-sticky-header="true"
            ng-reflect-theme="black-theme"
            ng-reflect-themes="[object Object],[object Object"
          >
            <div
              id="changeAnimType"
            />
            <div
              id="stikyHeader"
            />
            <div
              id="asideOpenMode"
            />
            <div
              id="autoNightMode"
            />
            <div
              id="from"
            />
            <div
              id="to"
            />
            <div
              id="animationEnabled"
            />
            <div
              id="changeTheme"
            />
          </app-aside>
        </div>
      </mat-sidenav>
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav-content
        class="mat-drawer-content mat-sidenav-content ng-tns-c21-0"
      >
        <div
          class="layout ng-tns-c21-0"
        >
          <div
            class="main"
          >
            <div
              class="main-content"
            >
              <app-header
                class="ng-tns-c21-0"
                ng-reflect-language=""
                ng-reflect-languages="[object Object],[object Object"
                ng-reflect-logged-in="false"
                ng-reflect-show-loader="false"
              >
                <div
                  id="lang"
                />
                <div
                  id="toggleNav"
                />
              </app-header>
              <div
                class="ng-tns-c21-0 ng-trigger ng-trigger-routeAnimation"
                style=""
              >
                <router-outlet
                  class="ng-tns-c21-0"
                />
              </div>
            </div>
            <app-footer
              class="ng-tns-c21-0"
            />
          </div>
        </div>
      </mat-sidenav-content>
      
    </mat-sidenav-container>
  </div>
</app-root>
`;

exports[`AppComponent should compile 1`] = `
<app-root
  asideOpenMode$={[Function Store]}
  asideOpenModes$={[Function Store]}
  autoNightMode$={[Function Store]}
  autoNightModeFrom$={[Function Store]}
  autoNightModeTo$={[Function Store]}
  effectiveTheme$={[Function Store]}
  language$={[Function Store]}
  languages$={[Function Store]}
  loggedIn$={[Function Store]}
  mainLoaderService={[Function MockMainLoaderService]}
  ngZone={[Function NgZone]}
  routeAnimateonsChangeType$={[Function Store]}
  routeAnimateonsChangeTypes$={[Function Store]}
  routeAnimationsEnabled$={[Function Store]}
  showMailLoader$={[Function Store]}
  sideNavOpen$={[Function Store]}
  stickyHeader$={[Function Store]}
  store={[Function Store]}
  theme$={[Function Store]}
  themes$={[Function Store]}
  user$={[Function Store]}
>
  <div
    class="ng-tns-c3-0 theme-wrapper default-theme sticky-header"
    ng-reflect-klass="theme-wrapper default-theme"
    ng-reflect-ng-class="[object Object]"
  >
    <mat-sidenav-container
      class="mat-drawer-container mat-sidenav-container"
    >
      
      <div
        class="mat-drawer-backdrop ng-star-inserted"
      />
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav
        class="mat-drawer mat-sidenav ng-tns-c5-1 ng-trigger ng-trigger-transform ng-tns-c3-0 mat-drawer-push ng-star-inserted"
        ng-reflect-mode="push"
        ng-reflect-opened="false"
        style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <app-aside
            class="ng-tns-c3-0"
            ng-reflect-aside-open-mode="push"
            ng-reflect-aside-open-modes="[object Object],[object Object"
            ng-reflect-auto-night-mode="false"
            ng-reflect-auto-night-mode-from="21"
            ng-reflect-auto-night-mode-to="7"
            ng-reflect-route-animateons-change-type="SLIDE"
            ng-reflect-route-animateons-change-types="[object Object],[object Object"
            ng-reflect-route-animations-enabled="true"
            ng-reflect-sticky-header="true"
            ng-reflect-theme="default-theme"
            ng-reflect-themes="[object Object],[object Object"
          >
            <div
              id="changeAnimType"
            />
            <div
              id="stikyHeader"
            />
            <div
              id="asideOpenMode"
            />
            <div
              id="autoNightMode"
            />
            <div
              id="from"
            />
            <div
              id="to"
            />
            <div
              id="animationEnabled"
            />
            <div
              id="changeTheme"
            />
          </app-aside>
        </div>
      </mat-sidenav>
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav-content
        class="mat-drawer-content mat-sidenav-content ng-tns-c3-0"
      >
        <div
          class="layout ng-tns-c3-0"
        >
          <div
            class="main"
          >
            <div
              class="main-content"
            >
              <app-header
                class="ng-tns-c3-0"
                ng-reflect-language=""
                ng-reflect-languages="[object Object],[object Object"
                ng-reflect-logged-in="false"
                ng-reflect-show-loader="false"
              >
                <div
                  id="lang"
                />
                <div
                  id="toggleNav"
                />
              </app-header>
              <div
                class="ng-tns-c3-0 ng-trigger ng-trigger-routeAnimation"
                style=""
              >
                <router-outlet
                  class="ng-tns-c3-0"
                />
              </div>
            </div>
            <app-footer
              class="ng-tns-c3-0"
            />
          </div>
        </div>
      </mat-sidenav-content>
      
    </mat-sidenav-container>
  </div>
</app-root>
`;

exports[`AppComponent should show aside 1`] = `
<app-root
  asideOpenMode$={[Function Store]}
  asideOpenModes$={[Function Store]}
  autoNightMode$={[Function Store]}
  autoNightModeFrom$={[Function Store]}
  autoNightModeTo$={[Function Store]}
  effectiveTheme$={[Function Store]}
  language$={[Function Store]}
  languages$={[Function Store]}
  loggedIn$={[Function Store]}
  mainLoaderService={[Function MockMainLoaderService]}
  ngZone={[Function NgZone]}
  routeAnimateonsChangeType$={[Function Store]}
  routeAnimateonsChangeTypes$={[Function Store]}
  routeAnimationsEnabled$={[Function Store]}
  showMailLoader$={[Function Store]}
  sideNavOpen$={[Function Store]}
  stickyHeader$={[Function Store]}
  store={[Function Store]}
  theme$={[Function Store]}
  themes$={[Function Store]}
  user$={[Function Store]}
>
  <div
    class="ng-tns-c15-0 theme-wrapper default-theme sticky-header"
    ng-reflect-klass="theme-wrapper default-theme"
    ng-reflect-ng-class="[object Object]"
  >
    <mat-sidenav-container
      class="mat-drawer-container mat-sidenav-container"
    >
      
      <div
        class="mat-drawer-backdrop ng-star-inserted"
      />
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
        tabindex="0"
      />
      <mat-sidenav
        class="mat-drawer mat-sidenav ng-tns-c17-1 ng-trigger ng-trigger-transform ng-tns-c15-0 mat-drawer-push ng-star-inserted ng-animating"
        ng-reflect-mode="push"
        ng-reflect-opened="true"
        style=""
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <app-aside
            class="ng-tns-c15-0"
            ng-reflect-aside-open-mode="push"
            ng-reflect-aside-open-modes="[object Object],[object Object"
            ng-reflect-auto-night-mode="false"
            ng-reflect-auto-night-mode-from="21"
            ng-reflect-auto-night-mode-to="7"
            ng-reflect-route-animateons-change-type="SLIDE"
            ng-reflect-route-animateons-change-types="[object Object],[object Object"
            ng-reflect-route-animations-enabled="true"
            ng-reflect-sticky-header="true"
            ng-reflect-theme="default-theme"
            ng-reflect-themes="[object Object],[object Object"
          >
            <div
              id="changeAnimType"
            />
            <div
              id="stikyHeader"
            />
            <div
              id="asideOpenMode"
            />
            <div
              id="autoNightMode"
            />
            <div
              id="from"
            />
            <div
              id="to"
            />
            <div
              id="animationEnabled"
            />
            <div
              id="changeTheme"
            />
          </app-aside>
        </div>
      </mat-sidenav>
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
        tabindex="0"
      />
      <mat-sidenav-content
        class="mat-drawer-content mat-sidenav-content ng-tns-c15-0"
      >
        <div
          class="layout ng-tns-c15-0"
        >
          <div
            class="main"
          >
            <div
              class="main-content"
            >
              <app-header
                class="ng-tns-c15-0"
                ng-reflect-language=""
                ng-reflect-languages="[object Object],[object Object"
                ng-reflect-logged-in="false"
                ng-reflect-show-loader="false"
              >
                <div
                  id="lang"
                />
                <div
                  id="toggleNav"
                />
              </app-header>
              <div
                class="ng-tns-c15-0 ng-trigger ng-trigger-routeAnimation"
                style=""
              >
                <router-outlet
                  class="ng-tns-c15-0"
                />
              </div>
            </div>
            <app-footer
              class="ng-tns-c15-0"
            />
          </div>
        </div>
      </mat-sidenav-content>
      
    </mat-sidenav-container>
  </div>
</app-root>
`;

exports[`AppComponent should sticky header 1`] = `
<app-root
  asideOpenMode$={[Function Store]}
  asideOpenModes$={[Function Store]}
  autoNightMode$={[Function Store]}
  autoNightModeFrom$={[Function Store]}
  autoNightModeTo$={[Function Store]}
  effectiveTheme$={[Function Store]}
  language$={[Function Store]}
  languages$={[Function Store]}
  loggedIn$={[Function Store]}
  mainLoaderService={[Function MockMainLoaderService]}
  ngZone={[Function NgZone]}
  routeAnimateonsChangeType$={[Function Store]}
  routeAnimateonsChangeTypes$={[Function Store]}
  routeAnimationsEnabled$={[Function Store]}
  showMailLoader$={[Function Store]}
  sideNavOpen$={[Function Store]}
  stickyHeader$={[Function Store]}
  store={[Function Store]}
  theme$={[Function Store]}
  themes$={[Function Store]}
  user$={[Function Store]}
>
  <div
    class="ng-tns-c18-0 theme-wrapper default-theme sticky-header"
    ng-reflect-klass="theme-wrapper default-theme"
    ng-reflect-ng-class="[object Object]"
  >
    <mat-sidenav-container
      class="mat-drawer-container mat-sidenav-container"
    >
      
      <div
        class="mat-drawer-backdrop ng-star-inserted"
      />
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav
        class="mat-drawer mat-sidenav ng-tns-c20-1 ng-trigger ng-trigger-transform ng-tns-c18-0 mat-drawer-push ng-star-inserted"
        ng-reflect-mode="push"
        ng-reflect-opened="false"
        style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <app-aside
            class="ng-tns-c18-0"
            ng-reflect-aside-open-mode="push"
            ng-reflect-aside-open-modes="[object Object],[object Object"
            ng-reflect-auto-night-mode="false"
            ng-reflect-auto-night-mode-from="21"
            ng-reflect-auto-night-mode-to="7"
            ng-reflect-route-animateons-change-type="SLIDE"
            ng-reflect-route-animateons-change-types="[object Object],[object Object"
            ng-reflect-route-animations-enabled="true"
            ng-reflect-sticky-header="true"
            ng-reflect-theme="default-theme"
            ng-reflect-themes="[object Object],[object Object"
          >
            <div
              id="changeAnimType"
            />
            <div
              id="stikyHeader"
            />
            <div
              id="asideOpenMode"
            />
            <div
              id="autoNightMode"
            />
            <div
              id="from"
            />
            <div
              id="to"
            />
            <div
              id="animationEnabled"
            />
            <div
              id="changeTheme"
            />
          </app-aside>
        </div>
      </mat-sidenav>
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav-content
        class="mat-drawer-content mat-sidenav-content ng-tns-c18-0"
      >
        <div
          class="layout ng-tns-c18-0"
        >
          <div
            class="main"
          >
            <div
              class="main-content"
            >
              <app-header
                class="ng-tns-c18-0"
                ng-reflect-language=""
                ng-reflect-languages="[object Object],[object Object"
                ng-reflect-logged-in="false"
                ng-reflect-show-loader="false"
              >
                <div
                  id="lang"
                />
                <div
                  id="toggleNav"
                />
              </app-header>
              <div
                class="ng-tns-c18-0 ng-trigger ng-trigger-routeAnimation"
                style=""
              >
                <router-outlet
                  class="ng-tns-c18-0"
                />
              </div>
            </div>
            <app-footer
              class="ng-tns-c18-0"
            />
          </div>
        </div>
      </mat-sidenav-content>
      
    </mat-sidenav-container>
  </div>
</app-root>
`;
